WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -26,7 +26,7 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
 
     private var cancellables: Set<AnyCancellable> = []
 
-    init(posModel: PointOfSaleAggregateModelProtocol,
+    init(posModel: PointOfSaleAggregateModel,
```

Comment:
Do you think the result should be that the aggregate model is only used within Views and not within ViewModels?

Reply:
@staskus Yes, using it in view models is pretty clunky, especially with this constraint.

If we move to `@Observable` this wouldn't be a hard-and-fast rule, but we may have to use `withObservationTracking` to be notified of changes. It's not as nice as Combine.

I don't expect that we'll have any ViewModels when I'm done with this, and I think it would be best if we could make any we do need in the relevant view's `init`, instead of passing them in. That will only work where we don't need the aggregate model in the viewmodel, as it gets a bit tricky with `EnvironmentObject` being the source for posModel.